qiskit/visualization/__init__.py

```diff
@@ -17,20 +17,160 @@
 
 .. currentmodule:: qiskit.visualization
 
+
+Overview
+========
+
+The visualization module contain functions that visualizes measurement outcome counts, quantum
+states, circuits, pulses, devices and more. 
+
+Prerequisites
+=============
+
+To use visualization functions, you are required to install visualization optionals to your
+development environment:
+
+.. code-block:: bash
+
+   pip install 'qiskit[visualization]'
+
 Counts and State Visualizations
 ===============================
 
+This section contains functions that visualize measurement outcome counts and quantum states.
+
+APIs
+----
+
+Counts Visualization
+^^^^^^^^^^^^^^^^^^^^
+
 .. autosummary::
    :toctree: ../stubs/
 
    plot_histogram
+
+State Visualization
+^^^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+   :toctree: ../stubs/
+
    plot_bloch_vector
    plot_bloch_multivector
    plot_state_city
    plot_state_hinton
    plot_state_paulivec
    plot_state_qsphere
 
+Example Usage
+-------------
+
+Counts Visualization
+^^^^^^^^^^^^^^^^^^^^
+
+Here is an example of using :func:`plot_histogram` to visualize measurement outcome counts:
+
+.. jupyter-execute::
+
+   from qiskit.tools.visualization import plot_histogram
+
+   counts = {"00": 501, "11": 499}
+   plot_histogram(counts)
+
+The data can be a dictionary with bit string as key and counts as value, or more commonly a
+:class:`~qiskit.result.Counts` object obtained from :meth:`~qiskit.result.Result.get_counts`.
+
```

Comment:
Should we add an example that uses a `QuantumCircuit` like you did in the State Visualization section?

Reply:
I think we don't need to do that. I think we either change the above examples to do backend.run and get_counts or not, but having two examples are redundant.

Reply:
You already have two examples of `plot_histogram` so it would indeed make more sense to change one of them. I think it would be nice to have at least one example in which `plot_histogram` is done with the usual Qiskit workflow (with some actual quantum involved) now that we've removed every reference to that from the code examples in #8351.

Reply:
I still think `get_counts` example doesn't belong here. If we need to add it, I would add it to `plot_histogram` function api page in #8351. Because the example involves too many things like `QuantumCircuit`, `BasicAer`, `transpile`, etc.:

```python
from qiskit import QuantumCircuit, BasicAer, transpile
from qiskit.visualization import plot_histogram

qc = QuantumCircuit(2, 2)
qc.h(0)
qc.cx(0, 1)
qc.measure([0, 1], [0, 1])

backend = BasicAer.get_backend('qasm_simulator')
tqc = transpile(qc, backend=backend)
job = backend.run(tqc)
counts = job.result().get_counts()

plot_histogram(counts)
```